### ISSUE TITLE
Possible Bug with assert

### ISSUE THREAD

**Issue by decentralisedkev** [stretchr/testify#747]
https://play.golang.org/p/vuatAwQZ7s8

When the following code is ran in the playground, the correct results are printed out.

The main function and failing tests have been included in the playground link.

**Comment by shaneHowearth** [comment]
So, I copied the code from the playground into a main.go and main_test.go, with the tests split off. I renamed the test functions such that I have TestRemove, TestRemoveOne, TestRemoveTwo. All the tests pass.

$ go version
go version go1.11 linux/amd64
$ go test -v
=== RUN   TestRemove
--- PASS: TestRemove (0.00s)
=== RUN   TestRemoveOne
--- PASS: TestRemoveOne (0.00s)
=== RUN   TestRemoveTwo
map[]
--- PASS: TestRemoveTwo (0.00s)
PASS
ok      brokenAssert    0.002s
